### PR TITLE
Fix underflow bug

### LINKED ISSUE #1426: uint256 sub can trigger an underflow
**Describe the bug**
The uint256_sub hint https://github.com/lambdaclass/cairo-vm/blob/722b2e29125ac5fda49cbd81d691941c485fa659/vm/src/hint_processor/builtin_hint_processor/uint256_utils.rs#L197-L217 can trigger an underflow panic

**To Reproduce**
Create a program using the hint [UINT256_SUB](https://github.com/lambdaclass/cairo-vm/blob/722b2e29125ac5fda49cbd81d691941c485fa659/vm/src/hint_processor/builtin_hint_processor/hint_code.rs#L319). 
```cairo
struct MyStruct0 {
	low: felt,
	high: felt,
}
func main() {
	let a = MyStruct0(low=54987052180710815841462937121160005261203577276882008045698301095581843457, high=10);
	let b = MyStruct0(low=2, high=3618502788666131213697322783095070105623107215331596699973087835080668217346);
	let res = MyStruct0(low=1, high=1);
	hint_func(a, b, res);
	return();
}
​
func hint_func(a: MyStruct0, b: MyStruct0, res: MyStruct0) -> (MyStruct0, MyStruct0, MyStruct0) {
	alloc_locals;
​
%{
def split(num: int, num_bits_shift: int = 128, length: int = 2):
    a = []
    for _ in range(length):
        a.append( num & ((1 << num_bits_shift) - 1) )
        num = num >> num_bits_shift
    return tuple(a)
​
def pack(z, num_bits_shift: int = 128) -> int:
    limbs = (z.low, z.high)
    return sum(limb << (num_bits_shift * i) for i, limb in enumerate(limbs))
​
a = pack(ids.a)
b = pack(ids.b)
res = (a - b)%2**256
res_split = split(res)
ids.res.low = res_split[0]
ids.res.high = res_split[1]
%}
	return(a, b, res);
}
```
Run the program and you will get a 
```
thread 'main' panicked at 'Cannot subtract b from a because b is larger than a.
```
panic

**Expected behavior**
The VM should return an error instead of panicking

**What version/commit are you on?**
v0.8.6

### PR DESCRIPTION
**Pull request by Juan-M-V**
# TITLE

## Description

Fixes #1426

## Checklist
- [x] Linked to Github Issue
- [x] Unit tests added
- [ ] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
  - [x] CHANGELOG has been updated.

### CODE REVIEW

**Comment by pefontana** [comment]
Nice one @Juan-M-V !!!
Can we add an integration test with a cairo program?

**Review by pefontana — COMMENTED** [review]
Nice one @Juan-M-V !!!
Can we add an integration test with a cairo program?

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 #### Upcoming Changes
 
+* fix: Using UINT256_HINT no longer panics when b is greater than 2^256 [#1430]([htt](https://github.com/lambdaclass/cairo-vm/pull/1430)
```
Comment (fmoletta) [inline CHANGELOG.md:5]:
```suggestion
* fix: Using UINT256_HINT no longer panics when b is greater than 2^256 [#1430](https://github.com/lambdaclass/cairo-vm/pull/1430)
```

**Comment by Juan-M-V** [comment]
> Nice one @Juan-M-V !!! Can we add an integration test with a cairo program?

I only added uint tests because normally the hint throws an error when b is too big. This happens on the python implementation. Now our VM throws an error instead of panicking.
Should I test the failure with a cairo program anyways?

File: CHANGELOG.md (hunk 1 above)
Comment (Juan-M-V) [reply CHANGELOG.md]: Resolved: https://github.com/lambdaclass/cairo-vm/pull/1430/commits/0cd3f140c519065727c71fbbe832d5dd194d5811

**Review by pefontana — APPROVED** [review]
Nice!

**Pull request merged**